lib/server.coffee

```diff
@@ -111,7 +113,6 @@ exports.request = (options = {}, outerCallback, onProgress) ->
 				progress(request(options))
 					.on('progress', onProgress)
 					.on('error', outerCallback)
-					.on('end', onProgress)
```

Comment:
Doesn't `close` get called on `error` as well, which would result in a double call there?